index.coffee

```diff
@@ -146,12 +147,21 @@ class ResumableRequest extends stream.Readable
 			@bytesTotal ?= parseInt(response.headers['content-length'])
 			@_reportProgress()
 			@_reportProgress.flush()
-			@response = cloneResponse(response).on('data', @push.bind(this))
+			@response = cloneResponse(response)
+			@destinations.forEach ([ dest, opts ]) =>
+				@response.pipe(dest, opts)
+			delete @destinations
 			@emit('response', @response)
 
-		# wrapping the stream in a v3 Readable stream streamlines the emission
-		# of some events and avoids some weird edge-cases
-		wrapLegacyStream(response).pipe(@response, end: false)
+		onData = @push.bind(this)
```

Comment:
If you use
```coffee
push: (data, encoding) =>
```
instead then you don't need to bind here, especially as `.bind` is pretty slow in node 6 compared to what `=>`/`_.bind` does

Reply:
Good point. Will update.